src/aurora/textureatlasfile.cpp

```diff
@@ -0,0 +1,111 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file
+ * A loader class for texture atlas xml files for dragonage and dragonage2
+ */
+
+#include "src/common/ustring.h"
+#include "src/common/readstream.h"
+#include "src/common/xml.h"
+#include "src/common/streamtokenizer.h"
+#include "src/common/strutil.h"
+
+#include "src/aurora/resman.h"
+#include "src/aurora/textureatlasfile.h"
+
+namespace Aurora {
+
+TextureAtlasFile::TextureAtlasFile(Common::UString name) {
+	Common::ScopedPtr<Common::SeekableReadStream> stream(ResMan.getResource(name, kFileTypeXML));
+	load(*stream);
+}
+
+TextureAtlasFile::TextureAtlasFile(Common::SeekableReadStream &stream) {
+	load(stream);
+}
+
+bool TextureAtlasFile::isAtlasTexture(const Common::UString &texture) const {
+	return _atlasTextures.find(texture) != _atlasTextures.end();
+}
+
+void TextureAtlasFile::getAtlasTexture(const Common::UString &texture,
+                                       Common::UString &textureFile, float &x, float &y, float &w, float &h) const {
+	std::map<Common::UString, AtlasTexture>::const_iterator iter = _atlasTextures.find(texture);
+	if (iter == _atlasTextures.end()) {
+		throw Common::Exception("Atlas Texture %s not found", texture.c_str());
+	}
+
+	const AtlasTexture &atlasTexture = iter->second;
+	textureFile = atlasTexture.textureFile;
+	x = atlasTexture.x;
+	y = atlasTexture.y;
+	w = atlasTexture.w;
+	h = atlasTexture.h;
+}
+
+void TextureAtlasFile::load(Common::SeekableReadStream &stream) {
+	Common::XMLParser parser(stream);
+
+	const Common::XMLNode &node = parser.getRoot();
+
+	if(node.getName() != "AtlasData")
```

Comment:
Since you're comparing strings here, you should probably do these checks case-insensitively. Just so that "ATLASDATA" won't mess you up.

In fact, you can just give the `Common::XMLParser` a true for the second parameter (`makeLower`, and yes, that should probably be better documented), and then just do all string comparisons with the lowercase version (i.e. "atlasdata", in this case).

Reply:
But every possible atlas is writing them as "AtlasData". So I do not see the point in making this check less strict when it already covers all possible atlas files.

Reply:
Eh, fair enough. We'll see if there's mods that depends on the game being case-insensitive there. And yeah, I'm not even sure the game is. :P

Reply:
At least i can assure you, that the game does it. I iterated over all atlas xml files with this code and there never was a problem.